src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,315 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.TextWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP
+namespace Microsoft.OData.Json
+{
+    using System;
+    using System.Buffers;
+    using System.IO;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    internal sealed partial class ODataUtf8JsonWriter
+    {
+        /// <summary>
+        /// Starts a TextWriter value scope with the specified content type.
+        /// </summary>
+        /// <param name="contentType">The content type of the TextWriter value scope.</param>
+        /// <returns>A TextWriter for writing JSON data.</returns>
+        public TextWriter StartTextWriterValueScope(string contentType)
+        {
+            this.WriteSeparatorIfNecessary();
+            this.currentContentType = contentType;
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            }
+
+            this.Flush();
+
+            return new ODataUtf8JsonTextWriter(this);
+        }
+
+        /// <summary>
+        /// Ends a TextWriter value scope.
+        /// </summary>
+        public void EndTextWriterValueScope()
+        {
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+                this.Flush();
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously starts a TextWriter value scope with the specified content type.
+        /// </summary>
+        /// <param name="contentType">The content type of the TextWriter value scope.</param>
+        /// <returns>A task representing the asynchronous operation. The task result is a TextWriter for writing JSON data.</returns>
+        public async Task<TextWriter> StartTextWriterValueScopeAsync(string contentType)
+        {
+            this.WriteSeparatorIfNecessary();
+            this.currentContentType = contentType;
+            if (!IsWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            }
+
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+
+            return new ODataUtf8JsonTextWriter(this);
+        }
+
+        /// <summary>
+        /// Asynchronously ends a TextWriter value scope.
+        /// </summary>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public async Task EndTextWriterValueScopeAsync()
+        {
+            if (!IsWritingJson)

```

Comment:
Do the tests verify how different content types are handled?

Reply:
I've added the `text/html` and `text/plain`. I'm not sure about the `application/json` one..

Reply:
It would be good to be sure. Since `application/json` and null/empty string is something we explicitly check for in the `IsWritingJson` property, then we should have explicit tests for `application/json` and for null/empty content types and ensure the quotes are written as expected.